tools/nni_trial_tool/web_channel.py

```diff
@@ -16,19 +16,23 @@ def __init__(self, args):
         self.args = args
         self.client = None
         self.in_cache = b""
+        self.timeout = 10
 
         super(WebChannel, self).__init__(args)
 
         self._event_loop = None
 
     def _inner_open(self):
         url = "ws://{}:{}".format(self.args.nnimanager_ip, self.args.nnimanager_port)
-        nni_log(LogType.Info, 'WebChannel: connected with info %s' % url)
-
-        connect = websockets.connect(url)
-        self._event_loop = asyncio.get_event_loop()
-        client = self._event_loop.run_until_complete(connect)
-        self.client = client
+        try:
+            connect = asyncio.wait_for(websockets.connect(url), self.timeout)
+            self._event_loop = asyncio.get_event_loop()
+            client = self._event_loop.run_until_complete(connect)
```

Comment:
Is it possible to hang at this line in some cases? Is a timeout needed for event loop?

Reply:
This self._event_loop.run_until_complete(connect) is for connect method, and connect has already handled timeout.